src/components/sidebar/tabs/NodeParamSideBarTab.vue

```diff
@@ -0,0 +1,329 @@
+<template>
+  <Accordion value="0">
+    <!-- gallery: maybe will merge to other sidepanel -->
+    <!-- <AccordionPanel value="carousel">
+      <AccordionHeader>
+        <i class="pi pi-image" />
+        Gallery
+      </AccordionHeader>
+      <AccordionContent>
+        <p class="m-0">
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
+          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+          aliquip ex ea commodo consequat. Duis aute irure dolor in
+          reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
+          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+          culpa qui officia deserunt mollit anim id est laborum.
+        </p>
+      </AccordionContent>
+    </AccordionPanel> -->
+    <SelectButton
+      class="custom-select-button"
+      v-model="selectedCategory"
+      :options="nodeParamStore.categories"
+      optionLabel="name"
+      @update:modelValue="(newValue) => onSelectCata(newValue)"
+      aria-labelledby="basic"
+    />
+    <!-- nodes -->
+    <AccordionPanel
+      v-for="(node, index) in nodeParamStore.filterNodes"
+      :key="index"
+      :value="node.id.toString()"
+    >
+      <AccordionHeader
+        :class="[
+          'custom-header',
+          { bypass: node.mode === 2 || node.mode === 4 }
+        ]"
+        :style="{ backgroundColor: node.color }"
+      >
+        <ToggleButton
+          @click.stop
+          v-model="node.bypass"
+          onIcon="pi pi-power-off"
+          offIcon="pi pi-power-off"
+          class="custom-toggle"
+          @update:modelValue="(newValue) => onBypass(node.id, newValue)"
+        />
+        {{ node.title }}
+      </AccordionHeader>
+      <AccordionContent
+        :class="[
+          'custom-content',
+          { bypass: node.mode === 2 || node.mode === 4 }
+        ]"
+        :style="{ backgroundColor: node.bgcolor }"
+      >
+        <template v-for="(widget, index) in node.widgets" :key="index">
```

Comment:
Can you map different widget types to different components with `<component :is="foo"/>`. You can refer to the `SettingGroup` https://github.com/Comfy-Org/ComfyUI_frontend/blob/d5f30be06d8ce4369f12ef22e27779a5cb9b0423/src/components/dialog/content/setting/SettingGroup.vue#L19-L27

Reply:
BTW, there is an input slider component which is a synced slider with input box that you can reuse: https://github.com/Comfy-Org/ComfyUI_frontend/blob/main/src/components/dialog/content/setting/InputSlider.vue

You can lift it under `common/`